code/modules/antagonists/eldritch_cult/eldritch_effects.dm

```diff
@@ -202,20 +202,24 @@
 			cultie.current.client.images -= img
 		//clear the list
 		minds -= cultie
+	GLOB.reality_smash_track.smashes -= src
 	img = null
 	new /obj/effect/broken_illusion(drop_location())
 
 ///Makes the mind able to see this effect
 /obj/effect/reality_smash/proc/AddMind(var/datum/mind/cultie)
+	//dont runtime if any of these dont exist, we will get em next time.
 	if(cultie.current.client)
-		minds |= cultie
 		cultie.current.client.images |= img
+		minds |= cultie
+
 
 ///Makes the mind not able to see this effect
 /obj/effect/reality_smash/proc/RemoveMind(var/datum/mind/cultie)
-	if(cultie.current.client)
-		minds -= cultie
+	if(cultie?.current?.client)
 		cultie.current.client.images -= img
+		minds -= cultie
```

Comment:
Will the mind reference persist in the list if the client has DCed?

Reply:
if the client has DC'd the client becomes a null, but since we have a registered MOB_LOGIN linked to the mob, if they disconnect they will be put back onto the list and the old entry will be deleted

Reply:
Right, as we talked on the coding channel, this tracks minds, not clients.
Minds persist through client disconnections unless they are associated to a `new_player` mob, which shouldn't be the case here.

The list shouldn't add and remove minds based on if there's a client or not.
It should register and unregister to logins and logoffs instead, when adding or removing minds to the list.

Reply:
Wait lemme rephrase the way it works. We add mind using AddMind() now reality_smash_tracker is registered to the mobs login, if they logoff, their client in mind is nulled which causes the reality_smashes to not show up. When a client logs in again, it adds back alll needed images to their respective mind.client.images list. It doesnt remove anything from the list based on if there is a client or not.

Reply:
What your are pointing out now is the mind removal system, that happens when someone stops being a heretic and should no longer be able to see these.